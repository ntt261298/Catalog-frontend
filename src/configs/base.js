export default {
  apiHost: 'http://localhost:5000',
};
