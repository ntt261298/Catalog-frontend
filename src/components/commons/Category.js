import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import getCategories from '../../actions/category';

export class Category extends Component {
  componentDidMount() {
    this.props.getCategories();
  }

  render() {
    const { categories } = this.props;
    return (
      <div className="category">
        <h2>Catagories</h2>
        <ul>
          {
            categories.map(({ id, name }) => (
              <li key={id}>
                <Link to={`/category/${id}`}>{ name }</Link>
              </li>
            ))
          }
        </ul>
      </div>
    );
  }
}

Category.propTypes = {
  getCategories: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  categories: state.category.categories,
});

export default connect(mapStateToProps, { getCategories })(Category);
