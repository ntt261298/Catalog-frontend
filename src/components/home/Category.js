import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import getCategories from '../../actions/category';

class Category extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.getCategories();
  }

  render() {
    const { categories } = this.props;
    return (
      <div className="category">
        <h2>Catagories</h2>
        <ul>
          {
            categories.map(({ id, name }) => (
              <li key={id}>{ name }</li>
            ))
          }
        </ul>
      </div>
    );
  }
}

Category.propTypes = {
  getCategories: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf.isRequired,
};

const mapStateToProps = state => ({
  categories: state.category.categories,
});

export default connect(mapStateToProps, { getCategories })(Category);
