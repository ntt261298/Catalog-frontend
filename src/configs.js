import {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Enzyme config
configure({ adapter: new Adapter() });

// API host config
const defaultHost = 'http://localhost:5000';

export {
  shallow, mount, render, defaultHost,
};
