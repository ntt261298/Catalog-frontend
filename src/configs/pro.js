import base from 'configs/base';

export default {
  ...base,
};
