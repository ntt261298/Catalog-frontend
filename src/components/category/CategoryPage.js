import React from 'react';
import { connect } from 'react-redux';
import '../../style/home.css';
import { PropTypes } from 'prop-types';
import Category from '../commons/Category';
import Item from '../commons/Item';
import Header from '../layout/Header';

export const CategoryPage = props => (
  <div>
    <Header />
    <div className="home-main">
      <Category />
      <Item type="category" id={props.match.params.id} />
    </div>
  </div>
);

CategoryPage.propTypes = {
  match: PropTypes.object.isRequired,
};

export default connect(null, null)(CategoryPage);
