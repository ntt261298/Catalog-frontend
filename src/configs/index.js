import {
  configure, shallow, mount, render,
} from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import dev from 'configs/dev';
import pro from 'configs/pro';

// Enzyme config
configure({ adapter: new Adapter() });

let configs;

if (process.env.REACT_APP_ENV === 'dev') {
  configs = dev;
} else if (process.env.REACT_APP_ENV === 'pro') {
  console.log('pro');
  configs = pro;
} else {
  // Default configs is dev
  configs = dev;
}

export { shallow, mount, render };

export default configs;
