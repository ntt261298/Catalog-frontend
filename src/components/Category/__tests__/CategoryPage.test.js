import React from 'react';
import { shallow } from 'configs';
import { CategoryPage } from 'components/Category/CategoryPage';

describe('components/Category/CategoryPage', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(
      <CategoryPage {...props} />,
    );
  };

  beforeEach(() => {
    props = {
      match: {
        params: {
          id: 1,
        },
      },
    };
  });

  it('should render Category page', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
