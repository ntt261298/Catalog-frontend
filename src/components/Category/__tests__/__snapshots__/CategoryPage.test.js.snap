// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Category/CategoryPage should render Category page 1`] = `
<div>
  <Connect(Component) />
  <div
    className="home-main"
  >
    <Connect(Category) />
    <Connect(Item)
      id={1}
      type="category"
    />
  </div>
</div>
`;
